runtime/lib/common/system_desc.cpp

```diff
@@ -93,6 +93,48 @@ getAllDeviceConnections(const vector<::tt::tt_metal::Device *> &devices) {
   return allConnections;
 }
 
+// Gather all physical cores by type for the device using metal device APIs
+static flatbuffers::Offset<::tt::target::ChipPhysicalCores>
+createChipPhysicalCores(const ::tt::tt_metal::Device *device,
+                        flatbuffers::FlatBufferBuilder &fbb) {
+
+  std::vector<::tt::target::Dim2d> worker_cores, dram_cores, eth_cores,
+      eth_inactive_cores;
+
+  auto logical_grid_size = device->logical_grid_size();
```

Comment:
`logical_grid_size` -> `compute_with_storage_grid_size`

Reply:
@nsmithtt  - Thx, okay, this is different than original idea in ticket, will change. But since `logical_grid_size()` in metal returned `worker_grid_size`, I think it makes sense to rename "workers" on our side to "compute" if we go with this, so we'd have:

```
compute
dram
eth
eth_inactive
```

Thoughts?

Reply:
On the fence about the rename, since we still have to call this API..

```
device->worker_core_from_logical_core()
```

Reply:
I think it's ok to leave them as worker, I'll leave it up to you.  The issue is that `logical_grid_size` returns the full size, which fast dispatch will take a row from. `compute_with_storage_grid_size` returns the cores you can actually use.

Reply:
got it, thx, updated to use new func, left as worker. resolved